R/grid_code_paths.R

```diff
@@ -373,6 +377,25 @@ tune_grid_loop_iter <- function(split,
 
   training <- rsample::analysis(split)
 
+  if (should_internal_split(workflow)) {
+    # if the workflow has a postprocessor that needs training (i.e. calibration),
+    # further split the analysis data into an "internal" analysis and
+    # assessment set.
+    # * the preprocessor and model (excluding the post-processor) are fitted
+    # on `analysis(split_post)`, the internal analysis set
+    # * that model generates predictions on `assessment(split_post)`, the
+    #   internal assessment set
+    # * the post-processor is trained on the predictions generated from the
+    #   internal assessment set
+    # * the model (including the post-processor) generates predictions on the
+    #   assessment set (not internal, i.e. `assessment(split)`) and those
+    #   predictions are assessed with performance metrics
+    split <- rsample::initial_split(training)
```

Comment:
I think that we should always do stratified splitting here. That might mean writing another function to do it since rsample was somewhat strict rules about how much data is needed to stratify.